Packs/Lumu/Integrations/Lumu/Lumu.py

```diff
@@ -0,0 +1,1080 @@
+            params=params,
+            headers=headers,
+        )
+
+        return response
+
+    def mute_incident_request(self, lumu_incident_id, comment):
+        params = assign_params(key=self.api_key)
+        data = {"comment": comment}
+        headers = self._headers
+
+        response = self._http_request(
+            "POST",
+            f"api/incidents/{lumu_incident_id}/mute",
+            params=params,
+            json_data=data,
+            headers=headers,
+        )
+
+        return response
+
+    def unmute_incident_request(self, lumu_incident_id, comment):
+        params = assign_params(key=self.api_key)
+        data = {"comment": comment}
+        headers = self._headers
+
+        response = self._http_request(
+            "POST",
+            f"api/incidents/{lumu_incident_id}/unmute",
+            params=params,
+            json_data=data,
+            headers=headers,
+        )
+
+        return response
+
+    def consult_incidents_updates_through_rest_request(self, offset, items, time):
+        params = assign_params(offset=offset, items=items, time=time, key=self.api_key)
+        headers = self._headers
+
+        response = self._http_request(
+            "GET",
+            "api/incidents/open-incidents/updates",
+            params=params,
+            headers=headers,
+        )
+
+        return response
+
+    def close_incident_request(self, lumu_incident_id, comment):
+        params = assign_params(key=self.api_key)
+        data = {"comment": comment}
+        headers = self._headers
+
+        response = self._http_request(
+            "POST",
+            f"api/incidents/{lumu_incident_id}/close",
+            params=params,
+            json_data=data,
+            headers=headers,
+        )
+
+        return response
+
+
+def get_hmac_sha256(key: Optional[str], comment: str):
+    if key is None:
+        key = ''
+    return hmac.new(key.encode(), comment.encode(), hashlib.sha256).hexdigest()
+
+
+def generate_hmac_sha256_msg(key: Optional[str], comment: str):
+    hmac_code = get_hmac_sha256(key, comment)
+    return f"{comment} hmacsha256:{hmac_code}"
+
+
+def validate_hmac_sha256(key: str, comment_w_hmac: str, separator="hmacsha256:"):
+    """
+
+    :param key:
+    :param comment_w_hmac: Resolved, from: ServiceNow Thu Nov 10 2022 07:56:13 GMT-0800 (PST),
+    close_code: Solution provided, close_notes: ddddddddddddd hmacsha256:
+    a9d7947047c371a10a44d07b381152951c46b0c954f8ddf1eeefc29fd120b8ae
+    :param separator:
+    :return:
+    """
+    result = [piece.strip() for piece in comment_w_hmac.split(separator)]
+    if len(result) != 2:
+        return False
+    comment, hash_mac = result
+    # logger.debug(f'{validate_hmac_sha256.__qualname__} comment: {comment}, {hash_mac}')
+    if get_hmac_sha256(key, comment) == hash_mac:
+        # logger.info(f'{validate_hmac_sha256.__qualname__} validating hmac is True: {comment}, {hash_mac}')
```

Comment:
```suggestion

```

Reply:
comment removed in commit 2bb633df7de2b53b93d8a8cdc1fcaef13e33c6a7